pylint/config/argument.py

```diff
@@ -39,10 +39,18 @@ def _yn_transformer(value: str) -> bool:
     )
 
 
+def _non_empty_string_transformer(value: str) -> str:
```

Comment:
Are we progressively discovering what we need to add based on the existing tests ? It's smart and efficient but I'm a little afraid that the tests will not be exhaustive. Maybe it would be safer to look at what optparse was actually doing and try to translate ?

Reply:
No we have this in `optparse` as well. This copied from `VALIDATORS`.

I just don't want to use anything in the old files because those are validators instead of transformers: ie., they get called every time a value is parsed or added instead of only when needed in the new setup.
Besides, I feel like it's easier to just cut-off everything instead of trying to build on top of it. Just copying what seems relevant :)

Reply:
Sorry, I misunderstood what you meant. 

Yeah, we're doing this progressively. But I think it's better to do it progressively now and do one final check at the end if we're missing something obvious. Copying stuff over from the `optparse` implementation is exactly what we want to avoid as I noticed a number of things are different in `argparse`. I'll do a big final check for this at the end.

Reply:
> Copying stuff over from the optparse implementation is exactly what we want to avoid 

Yeah, fair enough !